File: main_code/poc_kmp_sdks/ios/Sources/Platform/IOSPlatformServices.swift
import Foundation
import UIKit
import SharedSDK

/// iOS implementation of PlatformServices
/// Provides iOS-specific functionality
class IOSPlatformServices: PlatformServices {
    
    func getPlatformName() -> String {
        return "iOS \(UIDevice.current.systemVersion)"
    }
    
    func showMessage(message: String) {
        DispatchQueue.main.async {
            // In a real app, you might want to show a toast or alert
            // For now, we'll use a simple alert
            if let windowScene = UIApplication.shared.connectedScenes.first as? UIWindowScene,
               let window = windowScene.windows.first {
                let alert = UIAlertController(title: "Message", message: message, preferredStyle: .alert)
                alert.addAction(UIAlertAction(title: "OK", style: .default))
                window.rootViewController?.present(alert, animated: true)
            }
        }
    }
    
    func logMessage(tag: String, message: String) {
        print("[\(tag)] \(message)")
    }
    
    func getDeviceInfo() -> String {
        let device = UIDevice.current
        return "\(device.model) (\(device.systemName) \(device.systemVersion))"
    }
}

File: main_code/poc_kmp_sdks/shared/src/iosMain/kotlin/main_code/platform/IOSPlatformServices.kt
package main_code.platform

import platform.Foundation.NSLog
import platform.UIKit.UIDevice

/**
 * iOS implementation of PlatformServices
 * Provides iOS-specific functionality
 */
class IOSPlatformServices : PlatformServices {
    
    override fun getPlatformName(): String {
        return "iOS ${UIDevice.currentDevice.systemVersion}"
    }
    
    override fun showMessage(message: String) {
        // TODO: Implement iOS-specific message display (e.g., UIAlertController)
        // For now, just log the message
        NSLog("Message: $message")
    }
    
    override fun logMessage(tag: String, message: String) {
        NSLog("[$tag] $message")
    }
    
    override fun getDeviceInfo(): String {
        val device = UIDevice.currentDevice
        return "${device.model} (${device.systemName} ${device.systemVersion})"
    }
}